three-cards-divination/src/test/java/com/cyber/mysticism/tarot/controller/ThreeCardsDivinationControllerFunctionalTest.java

```diff
@@ -31,9 +32,10 @@ class ThreeCardsDivinationControllerFunctionalTest {
     void performReading_oneCard() throws Exception {
         Card card = new Card("The Fool", 0, "Major Arcana", "m00.jpg", List.of("Watch for new projects and new beginnings"));
         Map<String, Card> extracted = Map.of("past", card);
-        when(threeCardsDivinationService.getReading()).thenReturn(extracted);
+        when(threeCardsDivinationService.getReadingForUser("user", "email")).thenReturn(extracted);
```

Comment:
**Suggestion:** Consider using `@BeforeEach` to setup common test data and reduce code duplication. [best practice]
```suggestion
    Card card;
    Map<String, Card> extracted;

    @BeforeEach
    void setup() {
        card = new Card("The Fool", 0, "Major Arcana", "m00.jpg", List.of("Watch for new projects and new beginnings"));
        extracted = Map.of("past", card);
    }

    void performReading_oneCard() throws Exception {
        when(threeCardsDivinationService.getReadingForUser("user", "email")).thenReturn(extracted);
    }
```